File: solutions/1402-count-square-submatrices-with-all-ones/solution.java
class Solution {
    public int countSquares(int[][] matrix) {
        int sum1=0;
        for(int i=0;i<matrix.length;i++)
        {
            for(int j=0;j<matrix[i].length;j++)
            {
                if(i>0 && j>0 && matrix[i][j]==1)
                    matrix[i][j]+=Math.min(matrix[i-1][j-1],Math.min(matrix[i-1][j],matrix[i][j-1]));
                sum1+=matrix[i][j];    
            }
        }
        return sum1;
    }
}

File: solutions/1402-count-square-submatrices-with-all-ones/solution.py
class Solution(object):
    def countSquares(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: int
        """
        sum1=0;
        for i in range(len(matrix)):
            for j in range(len(matrix[i])):
                if(i>0 and j>0 and matrix[i][j]==1):
                    matrix[i][j]+=min(matrix[i-1][j-1],matrix[i-1][j],matrix[i][j-1])
                sum1+=matrix[i][j];    
        return sum1;
